Workflow for IWYU

PodushkaPIR opened the pull request:
# Include-What-You-Use (IWYU) Analysis Workflow

## Overview

This project provides a streamlined workflow for running Include-What-You-Use (IWYU) analysis on Desbordante configurable filtering and reporting. The tools help identify unnecessary includes and suggest missing ones.

## Prerequisites

### IWYU Installation

Choose one of the following installation methods:

#### Method 1: Package Installation
```bash
# Ubuntu/Debian
sudo apt install iwyu
```

#### Method 2: Build from Source

```bash
git clone https://github.com/include-what-you-use/include-what-you-use.git
cd include-what-you-use
mkdir build && cd build
cmake -G "Unix Makefiles" -DCMAKE_PREFIX_PATH=/usr/lib/llvm-19 ..
make
sudo make install
```

#### Required Versions:
- IWYU 0.23+
- Clang 19+


#### Additional Dependencies

- jq: JSON processor for configuration parsing
 
- Python 3: For output filtering script
 
- CMake: For generating build files


## Install dependencies on Ubuntu
```bash
sudo apt install jq python3 cmake
```

### File Descriptions
#### Core Scripts

- start_iwyu.sh - Main analysis script with command-line flags support

- output_cleaner.py - Filters and cleans IWYU output, removing noise and organizing results

#### Configuration Files

- iwyu_config.json - Central configuration for analysis parameters and filtering rules

- *.imp - IWYU mapping files [Format specification](https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/IWYUMappings.md)

p-senichenkov reviewed (commented):
I'll run these scripts, read IWYU docs and give more comments later.

p-senichenkov commented on iwyu/start_iwyu.sh:
```diff
@@ -0,0 +1,184 @@
+    if [ -n "$keep_files" ]; then
+        while read -r keep_file; do
+            keep_file=$(resolve_path "$keep_file")
+            args="$args -Xiwyu --keep=$keep_file"
```
I haven't read IWYU docs thoroughly yet, but looks like `-Xiwyu` is needed only when you pass options as `CXXFLAGS`.

PodushkaPIR committed:
Add infrastructure for Include-What-You-Use analysis

- run_iwyu.py - wrapper script for iwyu_tool with project-specific mapping files and compile_commands.json management
- mappings/ - mapping files to improve IWYU accuracy
